### PR TITLE
Fix compilation with MingW: NTSTATUS & PNTSTATUS were redefined.

### PR DESCRIPTION
**Pull request by jedisct1**
Once in uv-win.h, then in winapi.h

### CODE REVIEW

File: include/uv-private/uv-win.h [hunk 1]
```diff
@@ -134,6 +134,7 @@ typedef int (WSAAPI* LPFN_WSARECVFROM)
              LPWSAOVERLAPPED_COMPLETION_ROUTINE completion_routine);
 
 #ifndef _NTDEF_
+#define _NTDEF_ 1
```
Comment (bnoordhuis) [inline include/uv-private/uv-win.h:137]: That doesn't look right to me. @piscisaureus?

**Comment by luislavena** [comment]
@bnoordhuis @piscisaureus this is exactly the issue I described in #403

**Comment by bnoordhuis** [comment]
It is. But changing a `#ifndef` to `#define` doesn't seem like a proper fix to me.

**Comment by luislavena** [comment]
@bnoordhuis I see **adding** `#define` in the diff, to avoid the redefinition caused by winapi issue.

AFAIK is that inside `_NTDEF_` there are other things beyond `NTSTATUS`, and with this change we still have the duplicated code between the uv-internal and winapi headers.

**Comment by bnoordhuis** [comment]
Sorry, that's what I mean. That's why it doesn't look like a proper fix to me but I'll let @piscisaureus decide.

**Comment by piscisaureus** [comment]
Sorry for leaving this out there for so long, I was busy with other stuff.

**Comment by luislavena** [comment]
@piscisaureus we understand, no problem. Bring chocolate next time and we will forgive you :wink:

Please also see my comments on #403, both pull requests attempt to workaround the real issue: duplicated code around `NTSTATUS` in the headers.

Let us know your thoughts.

**Pull request closed without merging**